server/src/main/java/org/elasticsearch/common/logging/ThrottlingAndHeaderWarningLogger.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.logging;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.elasticsearch.common.SuppressLoggerChecks;
+
+/**
+ * This class wraps both <code>HeaderWarningLogger</code> and <code>ThrottlingLogger</code>
+ * which is a common use case across Elasticsearch
+ */
+public class ThrottlingAndHeaderWarningLogger {
+    private final HeaderWarningLogger headerWarningLogger = new HeaderWarningLogger();
+    private final ThrottlingLogger throttlingLogger;
+
+    public ThrottlingAndHeaderWarningLogger(Logger loggerName) {
+        this.throttlingLogger = new ThrottlingLogger(LogManager.getLogger(loggerName));
+    }
+
+    /**
+     * Logs a message, adding a formatted warning message as a response header on the thread context.
+     */
+    //TODO fix this
```

Comment:
apologies. It was meant to be a reminder to myself that ESLoggerUsageChecker has to be extended to verify the correct usage of this.
This can be used in code and someone might forget about passing all required parameters. So ESLoggerUsageChecker can detect this. 
```
logger.info("msg {} {}" ,1 ) ;// will fail
throttlingANdHeaderWarning.logAndAddWarning("msg {} {}, 1); // has to fail too
```